lib/board.rb

```diff
@@ -0,0 +1,64 @@
+class Board
+  attr_accessor :cells
+
+  def initialize
+    @cells = Array.new(9, " ")
+  end
+
+  def reset!
+    @cells = Array.new(9, " ")
+  end
+
+  def display
+    sep = "-----------"
+    puts " #{@cells[0]} | #{@cells[1]} | #{@cells[2]} "
+    puts sep
+    puts " #{@cells[3]} | #{@cells[4]} | #{@cells[5]} "
+    puts sep
+    puts " #{@cells[6]} | #{@cells[7]} | #{@cells[8]} "
+  end
+
+  def position(position)
+    position = position.to_i - 1
+    @cells[position]
+  end
+
+  def full?
+    @cells.all? { |position| position == "X" || position == "O" } ? true : false
+
+  end
+
+  def turn_count
+    counter = 0
+    @cells.each do |i|
+      if i == "X" || i == "O"
+        counter += 1
+      end
+    end
+    counter
```

Comment:
Use the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  cells.count {|cell| cell == "O" || cell == "X"}
end
```